apps/dash-lunar-lander/app.py

```diff
@@ -0,0 +1,435 @@
+import dash
+import dash_core_components as dcc
+import dash_html_components as html
+from dash.dependencies import Input, Output
+import numpy as np
+import casadi
+from casadi import SX, DM
+from math import cos, sin
+import plotly.graph_objects as go
+
+app = dash.Dash(__name__)
+server = app.server
+
+app.layout = html.Div(
+    children=[
+        html.Div(
+            children=[
+                html.Img(src=app.get_asset_url("dash-logo.png"), id="logo"),
+                html.H4("Dash Lunar Lander"),
+                html.A(
+                    id="github-link",
+                    children=["View source code on Github"],
+                    href="https://github.com/plotly/dash-sample-apps/blob/master/apps/dash-lunar-lander/app.py",
+                ),
+            ],
+            className="app-header",
+        ),
+        html.Div(
+            [
+                html.H3("Lunar Lander Trade Study App"),
+                html.P(
+                    "This app allows for exploring the trade space of a lunar lander. You can adjust different "
+                    "parameters on the lander, like max angular velocity and inital mass, using the sliders and a new "
+                    "optimal trajectory will be recomputed in near real time using Casadi."
+                ),
+                html.A(
+                    "The lunar lander model was based off of one in this paper,",
+                    href="https://arxiv.org/pdf/1610.08668.pdf",
+                    target="_blank",
+                ),
+                html.P(
+                    "The lunar lander model does not allowed for free rotation, instead requiring the lander to "
+                    "gimbal it’s engine and thrust to impart a rotation, giving the mass optimal control case a "
```

Comment:
change it's to its (sorry for being pedantic)

Reply:
all good, grammar is important!